splitio/sync/segment.py

```diff
@@ -230,17 +231,20 @@ def synchronize_segment(self, segment_name, till=None):
         try:
             fetched = self._read_segment_from_json_file(segment_name)
             if not self.segment_exist_in_storage(segment_name):
+                    self._segment_sha[segment_name] = self._get_sha(json.dumps(fetched))
```

Comment:
you are getting the `self._get_sha(json.dumps(fetched))` in several places over an over. Maybe you can get sha at the beginning

Reply:
done